### PR TITLE
[kotlin][client] gson complete integration

### PR DESCRIPTION
**Pull request by 4brunu**
The kotlin client generator, has an option to choose gson as the serialization library, but this only affects the models, the ApiClient still uses Moshi despite choosing gson.
This PR propagates the gson usage to everywhere when it's the choosen serialization library.
It fixes https://github.com/OpenAPITools/openapi-generator/issues/4331.

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@jimschubert (2017/09) ❤️, @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10)

### CODE REVIEW

**Comment by wing328** [comment]
I ran `gradle test` with the `kotlin-client-gson` client and got the following errors:
```
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :compileKotlin
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ByteArrayAdapter.kt: (6, 12): Unresolved reference: json
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/ByteArrayAdapter.kt: (24, 13): Unresolved reference: JSONObject
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/DateAdapter.kt: (6, 12): Unresolved reference: json
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/DateAdapter.kt: (28, 13): Unresolved reference: JSONObject
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/LocalDateAdapter.kt: (6, 12): Unresolved reference: json
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/LocalDateAdapter.kt: (26, 13): Unresolved reference: JSONObject
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/LocalDateTimeAdapter.kt: (6, 12): Unresolved reference: json
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/LocalDateTimeAdapter.kt: (26, 13): Unresolved reference: JSONObject
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/UUIDAdapter.kt: (6, 12): Unresolved reference: json
e: /Users/williamcheng/Code/openapi-generator/samples/client/petstore/kotlin-gson/src/main/kotlin/org/openapitools/client/infrastructure/UUIDAdapter.kt: (25, 13): Unresolved reference: NULL

> Task :compileKotlin FAILED

FAILURE: Build failed with an exception.

```

Does it work for you locally?

**Comment by wing328** [comment]
Can you please add an entry below https://github.com/OpenAPITools/openapi-generator/blob/master/pom.xml#L1253 to test the kotlin-client-gson samples÷

**Comment by 4brunu** [comment]
@wing328 the tests are now passing in the CI server. 
Can you please review this PR? 
Thanks 👍

File: bin/kotlin-client-okhttp3.sh [hunk 1]
```diff
@@ -30,5 +30,3 @@ export JAVA_OPTS="${JAVA_OPTS} -Xmx1024M -DloggerPath=conf/log4j.properties"
 ags="generate -t modules/openapi-generator/src/main/resources/kotlin-client -i modules/openapi-generator/src/test/resources/2_0/petstore.yaml -g kotlin --artifact-id kotlin-petstore-okhttp3 --library jvm-okhttp3 -o samples/client/petstore/kotlin-okhttp3 $@"
 
 java ${JAVA_OPTS} -jar ${executable} ${ags}
-
-#cp CI/samples.ci/client/petstore/kotlin-okhttp3/pom.xml samples/client/petstore/kotlin-okhttp3/pom.xml
```
Comment (wing328) [inline bin/kotlin-client-okhttp3.sh]: Shall we undo this change?

File: bin/kotlin-client-nonpublic.sh [hunk 2]
```diff
@@ -30,5 +30,3 @@ export JAVA_OPTS="${JAVA_OPTS} -Xmx1024M -DloggerPath=conf/log4j.properties"
 ags="generate -t modules/openapi-generator/src/main/resources/kotlin-client -i modules/openapi-generator/src/test/resources/2_0/petstore.yaml -g kotlin --artifact-id kotlin-petstore-nonpublic --additional-properties nonPublicApi=true -o samples/client/petstore/kotlin-nonpublic $@"
 
 java ${JAVA_OPTS} -jar ${executable} ${ags}
-
-cp CI/samples.ci/client/petstore/kotlin-string/pom.xml samples/client/petstore/kotlin-string/pom.xml
```
Comment (wing328) [inline bin/kotlin-client-nonpublic.sh]: Shall we undo this change?

Comment (wing328) [reply bin/kotlin-client-nonpublic.sh]: No worry. I'l update those in another PR.

File: modules/openapi-generator/src/main/resources/kotlin-client/build.gradle.mustache [hunk 3]
```diff
@@ -46,5 +46,5 @@ dependencies {
     {{#threetenbp}}
     compile "org.threeten:threetenbp:1.4.0"
     {{/threetenbp}}
-    testCompile "io.kotlintest:kotlintest-runner-junit5:3.4.2"
+    testCompile "io.kotlintest:kotlintest-runner-junit5:3.1.0"
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/kotlin-client/build.gradle.mustache]: @4brunu so downgrading the version actually fixes the issue...

Comment (4brunu) [reply modules/openapi-generator/src/main/resources/kotlin-client/build.gradle.mustache]: Yes
